examples/threading/threading.c

```diff
@@ -4,17 +4,52 @@
 #include <stdio.h>
 
 // Optional: use these functions to add debug or error prints to your application
-#define DEBUG_LOG(msg,...)
-//#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
+//#define DEBUG_LOG(msg,...)
+#define DEBUG_LOG(msg,...) printf("threading: " msg "\n" , ##__VA_ARGS__)
 #define ERROR_LOG(msg,...) printf("threading ERROR: " msg "\n" , ##__VA_ARGS__)
 
 void* threadfunc(void* thread_param)
 {
 
     // TODO: wait, obtain mutex, wait, release mutex as described by thread_data structure
     // hint: use a cast like the one below to obtain thread arguments from your parameter
-    //struct thread_data* thread_func_args = (struct thread_data *) thread_param;
-    return thread_param;
+    struct thread_data* thread_func_args = (struct thread_data *) thread_param;
+
+    int ret = usleep((thread_func_args->wait_obtain_ms)*1000);
+    if (ret != 0)
+    {
+        ERROR_LOG("usleep failed");
+        thread_func_args->thread_complete_success = false;
+        return(thread_param);
+    }
+
+    ret = pthread_mutex_lock(thread_func_args->lock);
+    if (ret != 0)
+    {
+        ERROR_LOG("pthread_mutex_lock failed");
+        thread_func_args->thread_complete_success = false;
+        return(thread_param);
+    }
+
+    ret = usleep((thread_func_args->wait_release_ms)*1000);
+    if (ret != 0)
+    {
+        ERROR_LOG("usleep failed");
+        thread_func_args->thread_complete_success = false;
+        return(thread_param);
```

Comment:
@MuktaDarekar not safe to return here, since you've obtained the mutex lock above.

Reply:
done.